symengine/ruby/ext/symengine/ruby_basic.c

```diff
@@ -66,7 +66,7 @@ VALUE cbasic_eq(VALUE self, VALUE operand2) {
     basic_struct *this, *cbasic_operand2;
 
     Data_Get_Struct(self, basic_struct, this);
-    Data_Get_Struct(operand2, basic_struct, cbasic_operand2);
+    sympify(operand2, &cbasic_operand2);
```

Comment:
I have been getting [this](https://gist.github.com/abinashmeher999/7f4935459b334b7f8353) error, after the introduction of this method. I don't understand what has changed, except that I have passed the pointer to `cbasic_operand2`. The same thing(`Data_Get_Struct`) that happened earlier, happens in [`sympify`](https://github.com/sympy/symengine/pull/487/files#diff-5f89ca9c732a55713f6682f4f7a61d5eR4)